internal/scan-manager/core/src/test/java/org/wso2/security/tools/scanmanager/core/service/ScanEngineServiceImplTest.java

```diff
@@ -0,0 +1,213 @@
+        MockitoAnnotations.initMocks(this);
+        ScanManagerConfiguration.getInstance().initScanConfiguration(ScanMangerConfigurationBuilder.getConfiguration());
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new PowerMockObjectFactory();
+    }
+
+    @Test(dataProvider = "getScanEngineData", dataProviderClass = ServiceTestDataProvider.class)
+    public void testBeginPendingScans(Object scanDataObject, Object containerDataObject) {
+        try {
+            Scan scan = ServiceTestDataProvider.parseScanObject(scanDataObject);
+            Container container = ServiceTestDataProvider.parseContainerObject(containerDataObject);
+            if (scan != null && container != null) {
+                scan.setStatus(ScanStatus.SUBMIT_PENDING);
+                Mockito.when(scanService.getPendingScans(ScanStatus.SUBMIT_PENDING))
+                        .thenReturn(Collections.singletonList(scan));
+                Mockito.when(scanService.getByJobId(scan.getJobId())).thenReturn(scan);
+
+                Scan existingTestScan = new Scan();
+                existingTestScan.setScannerAppId(OCCUPIED_SCANNER_APP_ID);
+                Mockito.when(scanService.getByStatusesAndScannerAndProduct(new ArrayList<>(
+                                Arrays.asList(ScanStatus.SUBMITTED, ScanStatus.RUNNING, ScanStatus.CANCEL_PENDING)),
+                        scan.getScanner(), scan.getProduct())).thenReturn(Collections.singletonList(existingTestScan));
+
+                List<ScannerApp> testScannerAppList = Arrays.asList(new ScannerApp(scan.getScanner(),
+                                OCCUPIED_SCANNER_APP_ID, OCCUPIED_SCANNER_APP_ID, scan.getProduct()),
+                        new ScannerApp(scan.getScanner(), SCANNER_APP_ID, SCANNER_APP_NAME, scan.getProduct()));
+                Mockito.when(scannerService.getAppsByScannerAndAssignedProduct(scan.getScanner(),
+                        scan.getProduct())).thenReturn(testScannerAppList);
+                Mockito.when(scanService.update(Mockito.any(Scan.class))).thenReturn(scan);
+                Mockito.when(containerHandler.create(scan.getScanner().getImage(), ScanManagerConfiguration
+                                .getInstance().getScannerServiceHost(), ScanManagerConfiguration.getInstance()
+                                .getScannerServicePort(), container.getLabels(), new ArrayList<>(),
+                        new String[]{CONTAINER_ENV_NAME_SCAN_MANAGER_HOST + "=" + ScanManagerConfiguration
+                                .getInstance().getScanManagerHost(),
+                                CONTAINER_ENV_NAME_SCAN_MANAGER_PORT + "=" + ScanManagerConfiguration
+                                        .getInstance().getScanManagerPort()})).thenReturn(container);
+                Mockito.doNothing().when(containerHandler).start(container.getId());
+
+                PowerMockito.mockStatic(HTTPUtil.class);
```

Comment:
Would it be better to make the `HTTPUtil` a singleton and then not make methods `static`, rather than bringing in `PowerMockito`. 

`HTTPUtil` will also become testable if we do so.

Reply:
Even if we make the HTTPUtill a singleton, the Mockito will be unable to create a mock instance of  HTTPUtil (due to the private constructor) without the support of PowerMockito.

Reply:
Noted